realm/realm-library/src/main/java/io/realm/FieldAttribute.java

```diff
@@ -33,7 +33,10 @@
      * Marks a field as a primary key. This also implicitly mark it as {@link #INDEXED} and {@link #REQUIRED}.
      *
      * @see io.realm.annotations.PrimaryKey
+     * @deprecated adding or removing primary key after object schema gets created will be disallowed in
+     * realm-java 4.0.0.
      */
+    @Deprecated
```

Comment:
I am not quite sure it is good to mark this as `@Deprecated`, currently the only public API using this is `addField()`, so removing this in the 4.0.0 is accepted? Or do we need this in the future API like `RealmObjectSchema.getFieldAttribute()`?